server/src/main/java/org/elasticsearch/cluster/metadata/MappingMetadata.java

```diff
@@ -57,6 +61,7 @@ public MappingMetadata(CompressedXContent mapping) {
         }
         this.type = mappingMap.keySet().iterator().next();
         this.routingRequired = routingRequired((Map<String, Object>) mappingMap.get(this.type));
+        this.fieldsForModels = null; // TODO: Set fieldsForModels here
```

Comment:
How would we go about setting `fieldsForModels` in this constructor?

Reply:
we can't. There's not enough information in the mapping for retrieving it.

We don't need that info to be present on those code paths though, so we're safe setting that to null / empty.